### DISCUSSION TITLE
Alternative ingredient

### DISCUSSION THREAD

**Discussion by bartkummel** [mealie-recipes/mealie#3717] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Some recipes mention alternative ingredients in the list, like:
- ingredient A
- ingredient B
- ingredient C or ingredient D

Currently, there is no elegant way to enter such a recipe in Mealie.

### Please provide a concise description of the feature that would resolve your issue.

- We need a way to mark ingredients as alternatives to each other. When inserting an ingredient, there could be a checkbox "is alternative to", when checked, it could show a drop down list of all ingredients that are already on the list.
- When rendering the recipe, there should be some nice markup/layout that that makes it really clear that you can choose ingredient C _OR_ ingredient D.
- When adding the ingredient to a shopping list, only one of the alternatives should be selected. If you select a different alternative, the other(s) should be unselected; like a radio button.

### Please consider and list out some caveats or tradeoffs made in your design decision

- It makes entering ingredients a bit more complicated. But I think that –if you don't use the "alternatives" feature– the added complexity is minimal. You could make this an optional feature to mitigate this.
- It might be hard to find a good layout for the read only / cooking view.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by niedon** [comment]
This request and marking ingredients as optional would make the Recipe Finder the best tool for finding what to cook based on what you have on hand.

**Comment by DesignerThan95** [comment]
I agree. Alternative ingredients would be amazing!
